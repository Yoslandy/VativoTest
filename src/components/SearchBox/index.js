import React, { useState, useCallback } from "react"
import PropTypes from "prop-types"
import Icon from "../Icon"

export default function SearchBox(props) {
    const { value, className, onChange } = props
    const [state, setState] = useState(value)
    const [focus, setFocus] = useState(false)

    const onChangeLocal = useCallback(
        event => {
            setState(event.target.value)
            onChange(event.target.value)
        },
        [state]
    )

    const onFocus = () => {
        setFocus(true)
    }

    const onBlur = () => {
        setFocus(false)
    }

    return (
        <div
            className={`bg-white rounded-md w-full flex items-center hover:text-white hover:bg-primary-100 ${
                focus && "bg-primary-100"
            } ${className}`}
        >
            <Icon type="search" className="p-2" />
            <input
                onFocus={onFocus}
                onBlur={onBlur}
                type="text"
                className="input-focus-border-none w-full border-0 p-0 bg-transparent hover:placeholder:text-white"
                onChange={onChangeLocal}
                placeholder="Search"
            />
        </div>
    )
}

SearchBox.propTypes = {
    value: PropTypes.string,
    className: PropTypes.string,
    onChange: PropTypes.func,
}

SearchBox.defaultProps = {
    value: "",
    className: "",
    onChange: () => {},
}
