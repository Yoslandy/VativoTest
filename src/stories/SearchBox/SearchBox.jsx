import React from "react"
import SearchBox from "../../components/SearchBox"

export default function SearchBoxCustom() {
    return <SearchBox />
}
