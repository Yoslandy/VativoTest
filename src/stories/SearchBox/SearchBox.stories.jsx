import React from "react"
import "../../assets/styles/build.css"
import SearchBox from "./SearchBox"

export default {
    title: "Example/SearchBox",
    component: SearchBox,
}

function Template() {
    return <SearchBox />
}

export const Default = Template.bind({})
