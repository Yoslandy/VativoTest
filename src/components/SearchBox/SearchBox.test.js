import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import SearchBox from "./index"

test("Renders SearchBox form correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <SearchBox />
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
