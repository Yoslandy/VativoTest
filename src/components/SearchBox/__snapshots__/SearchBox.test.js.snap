// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SearchBox form correctly 1`] = `
<div
  className="bg-white rounded-md w-full flex items-center  hover:text-white"
>
  <span
    className="material-icons p-2"
    onClick={[Function]}
    role="button"
  >
    search
  </span>
  <input
    className="input-focus-border-none w-full border-0 p-0 bg-transparent hover:placeholder:text-white"
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
</div>
`;
